### ISSUE TITLE
--target private-window doesn't work if no other qb instance is running

### ISSUE THREAD

**Issue by TheStutterNerd** [qutebrowser/qutebrowser#7880]
**Version info**:
```
qutebrowser v3.0.0
Git commit: 
Backend: QtWebEngine 6.5.2, based on Chromium 108.0.5359.220 (from api)
Qt: 6.5.2

CPython: 3.11.5
PyQt: 6.5.2

Qt wrapper info:
  PyQt6: success
  PyQt5: not imported
  -> selected: PyQt6 (via autoselect)

colorama: 0.4.6
jinja2: 3.1.2
pygments: 2.16.1
yaml: 6.0.1
adblock: 0.6.0
objc: no
PyQt6.QtWebEngineCore: 6.5.0
PyQt6.sip: 6.7.10
pdf.js: 3.7.107 (/usr/share/pdf.js/build/pdf.js)
sqlite: 3.43.0
QtNetwork SSL: OpenSSL 3.1.2 1 Aug 2023

Style: QFusionStyle
Platform plugin: xcb
OpenGL: Mesa, 4.5 (Compatibility Profile) Mesa 23.1.6-arch1.4
Platform: Linux-6.4.12-arch1-1-x86_64-with-glibc2.38, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.11/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt6, data path: /usr/share/qt6

Paths:
cache: /home/tsn/.cache/qutebrowser
config: /home/tsn/.config/qutebrowser
data: /home/tsn/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser

Autoconfig loaded: no
Config.py: /home/tsn/.config/qutebrowser/config.py has been loaded
Uptime: 0:00:01
```

**Does the bug happen if you start with `--temp-basedir`?**:
Yes

**Description**
`Qutebrowser --target private-window` doesn't work if no other qutebrowser instances are running. It opens the site in a normal instance instead. If a qutebrowser instance is already running, then it works fine.

In my case, I generally have 20+ tabs open at all times and use private windows if I just temporarily want to check a website (to save myself the load time, the resources, etc).

**How to reproduce**
1. Have no qutebrowser instance running
2. Run `qutebrowser --target private-window http://example.com`

**Comment by The-Compiler** [comment]
The `--help` output for `--target` says:

> How URLs should be opened if there is already a qutebrowser instance running.

So this seems expected. Any other `--target` option would make no sense when there is no window open yet. Perhaps a `--private` flag would make sense, but I see this as a feature request.

Please indeed open a separate issue for the `-R` thing.

**Comment by TheStutterNerd** [comment]
Ah yeah I did notice it said that. 

Sorry I should have explained that more. I just presumed `--target private-window` would just open a completely unlinked window/session like how it does on all browsers. I use it a lot.

I didn't see any other option to run a private instance so I just thought thats probably it. All of the other options I expected to open in a normal window but the private-window one I thought it'd just run a completely seperate instance unrelated to the normal one. It seemed weird seeing it in the --target args relating the needing an instance running because why would it? Sort of thing.

Thanks though for explaining. I legit had no clue you couldn't, otherwise I would have done it as a feature request. I generally just leave qutebrowser open hence why I've only just realised. Cheers again though.

**Comment by Hund** [comment]
This is a bug, not a feature. :D The logic that qutebrowser uses for this, goes against the logic of every other browser.

I just assumed that this was either a bug or something that wasn't implemented yet, so I just wrote a shell script to make a ugly workaround for it. I eventually decided to see if anyone had reported this issue, which lead me to this ticket.

You should at least mention the `-R` flag in the description for the `--target` flag, if you don't want to use the same logic as everyone else. :)

**Comment by The-Compiler** [comment]
@Hund Can you elaborate? "It"? "logic of every other browser"?

**Comment by TheStutterNerd** [comment]
@The-Compiler I think he's just meaning that on every other browser, the instance/session is completely separate from a normal one like I explained.

For example, you can open private windows on vivaldi, chrome, edge, firefox, etc. All without having a normal instance/session already running.

**Comment by Hund** [comment]
> @The-Compiler I think he's just meaning that on every other browser, the instance/session is completely separate from a normal one like I explained.
> 
> For example, you can open private windows on vivaldi, chrome, edge, firefox, etc. All without having a normal instance/session already running.

This. :)

**Comment by The-Compiler** [comment]
Right, so apparently Chromium (with `--incognito`) and Firefox (with `--private-window`) only open that, and then once a new window gets opened (e.g. with ctrl-n), they load the previous session with all previously opened windows (and don't actually open a new window, funnily enough).

Seems rather surprising to me honestly, but can't say I've really thought about this much so far.
